src/VocabularyManager.Api/Controllers/VocabulariesController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using VocabularyManager.Core.Entities;
+using VocabularyManager.Core.Interfaces;
+using VocabularyManager.UseCases.DTOs;
+using VocabularyManager.UseCases.Interfaces;
+using VocabularyManager.Api.ActionFilters;
+using FluentValidation;
+using VocabularyManager.UseCases.Validators;
+
+namespace VocabularyManager.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class VocabulariesController : ControllerBase
+    {
+        private readonly IVocabularyStoreManager _vocabularyStoreManager;
+        private readonly IWordParser<WordDTO> _oxfordDictionaryParser;
+        public VocabulariesController(
+            IVocabularyStoreManager vocabularyStoreManager,
+            IWordParser<WordDTO> oxfordDictionaryParser)
+        {
+            _vocabularyStoreManager = vocabularyStoreManager;
+            _oxfordDictionaryParser = oxfordDictionaryParser;
+        }
+
+        [HttpGet]
+        public IActionResult GetVocabularyByUrlAsync([FromQuery] string url)
+        {
+            IAsyncEnumerable<WordDTO> vocabulary = _oxfordDictionaryParser
+                    .GetWordListByLinkAsync(url, new OxfordParsingWordDTOValidator());
+            return Ok(vocabulary);
+        }
+        [HttpPost]
+        [ServiceFilter(typeof(VocabularyValidationFilter))]
+        public async Task<IActionResult> CreateVocabulary(
+            [FromBody] Vocabulary vocabulary)
+        {
+            int vocabularyId = await _vocabularyStoreManager.CreateVocabulary(vocabulary);
+            return Ok(vocabularyId);
+        }
+        [HttpGet("all")]
+        public async Task<IActionResult> GetVocabularies()
+        {
+            return Ok(await _vocabularyStoreManager.GetVocabularies());
+        }
+        [HttpGet("byId")]
```

Comment:
this is a small problem, because that's not restful

you would have just `api/vocabularies/{vocabularyId}` as a path

for URL I would keep ``api/vocabularies` and combine getting all with url. when url it's provided, you give it for specific url, if not everything it's returned

what do you think?

Reply:
I think it makes sense according to swager documentation and side usage, it would be more readable)